ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -1212,8 +1241,16 @@ def uplink_peers(self):
 
         These are used to generate the "avd_topology_peers" fact covering downlinks for all devices.
         '''
-        if self.uplinks is not None:
-            return [uplink['peer'] for uplink in self.uplinks]
+        fabric_name = get(self._hostvars, "fabric_name", required=True)
+        inventory_group = get(self._hostvars, f"groups.{fabric_name}", required=True)
+        uplink_switches = default(self.uplink_switches, [])
+        uplink_peers = []
+        for uplink_switch in uplink_switches:
+            if uplink_switch is None or uplink_switch not in inventory_group:
+                # Invalid uplink_switch. Skipping.
+                continue
+            uplink_peers.append(uplink_switch)
```

Comment:
```suggestion
        uplink_peers = [ uplink_switch for uplink_switch in uplink_switches if (uplink_switch is not None and uplink_switch in inventory_group)]
```

Which I guess could be simplified further with
```
        uplink_peers = [ uplink_switch for uplink_switch in uplink_switches if uplink_switch in inventory_group]
```

given that `None` should never be in inventory_group

Reply:
agreed, I will use the list comprehension more going forward, and yes, `None` should not be in the inventory_group :)